subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -135,19 +138,24 @@ public DefaultJvmTestSuite(String name, ConfigurationContainer configurations, D
 
         addDefaultTestTarget();
 
+        // Until the values here can be finalized upon the user setting them (see the org.gradle.api.tasks.testing.Test#testFramework(Closure) method),
+        // in Gradle 8, we will be executing the provider lambda used as the convention multiple times.  So make sure, within a Test Suite, that we
+        // always return the same one via computeIfAbsent() against this map.
+        final Map<Frameworks, TestFramework> testFrameworkMap = new HashMap<>(3);
+
         this.targets.withType(JvmTestSuiteTarget.class).configureEach(target -> {
             target.getTestTask().configure(task -> {
                 task.getTestFrameworkProperty().convention(getTestingFramework().map(framework -> {
                     switch(framework.framework) {
```

Comment:
nit: maybe we can brainstorm a better lambda parameter name. `framework.framework` is kind of icky.

But I really like the `map.computeIfAbsent` below.

Reply:
Good idea.  Its even work in the top of this file, where we have a `TestingFramework` which contains a `Frameworks` instance that we use to create the `TestFramework`.  I tried some renaming, and think it looks a little clearer now.